langchain4j-open-ai/src/main/java/dev/langchain4j/model/openai/OpenAiStreamingResponseBuilder.java

```diff
@@ -29,7 +29,7 @@ public class OpenAiStreamingResponseBuilder {
     private final StringBuffer toolNameBuilder = new StringBuffer();
     private final StringBuffer toolArgumentsBuilder = new StringBuffer();
 
-    private final Map<Integer, ToolExecutionRequestBuilder> indexToToolExecutionRequestBuilder = new HashMap<>();
```

Comment:
sorry, I found `toolCall.index()` is null, the specific reason is still unclear

Reply:
Does this happen with OpenAI? Every model?

Reply:
I am unable to access OpenAI in my region, so I am using a third-party proxy service 😿
I cannot confirm if it is a third-party service issue 😿

Reply:
I see. So `toolCall.index() == null` blocks you from testing changes in this PR?

Reply:
I change `indexToToolExecutionRequestBuilder` to `HashMap` to ignore `toolCall.index() == null`
The `StreamingAiServicesWithToolsWithoutMemoryIT` in PR is passed

Reply:
@Kugaaa this will most probably not work when there are multiple tool calls in the LLM response.

`ToolCall toolCall = delta.toolCalls().get(0)` means that there is only one `ToolCall` **_in the current chunk_** `ChatCompletionResponse`. But if LLM decides to call multiple tools in the same response, `ToolCall toolCall = delta.toolCalls().get(0)` will return another `ToolCall`, with other name/arguments. I guess we need to count `index` manually or find other solution (maybe rely on `toolCall.id()`, but some models probably don't return it?)

Reply:
Yes, I understand its purpose. I modified it to a `HashMap` for testing the `StreamingAiServicesWithToolsWithoutMemoryIT.should_execute_multiple_tools_sequentially_then_answer` method (since there would only be one `ToolCall` in this case). 
However, I accidentally committed the code, so I reverted it in [this commit](https://github.com/langchain4j/langchain4j/pull/1280/commits/5ff6e9cfb0d7fc37ecc1f753950c1d331d6765bb)(I am not trying to express that I want to replace `ConcurrentHashMap` with `HashMap`).

Additionally, I checked the [OpenAI doc](https://platform.openai.com/docs/api-reference/chat/streaming), and there should be the `index` field. 
I have already raised an [issue](https://github.com/chatanywhere/GPT_API_free/issues/254#issuecomment-2180471421) with the third-party service I'm using to inquire about why this is happening. I suspect it's an issue with the third-party service.